File: docs/DSA/src/py/binary_search.py
class Solution:
    def search(self, nums: list[int], target: int) -> int:
        l, r = 0, len(nums)
        while l < r:
            mid = l + (r-l)//2
            if nums[mid] >= target:
                r = mid
            else:
                l = mid+1

        return l if l < len(nums) and nums[l] == target else -1

File: DSA/src/go/binary_search.go
package main

func search(nums []int, target int) int {
	left, right := 0, len(nums)
	var mid int
	for left < right {
		mid = left + (right-left)/2
		if nums[mid] >= target {
			right = mid
		} else {
			left = mid + 1
		}
	}
	if left < len(nums) && nums[left] == target {
		return left
	}
	return -1
}
